### PR TITLE
fix: convert cids to v1 when browsing dag in cleanBlockStore function

### PR DESCRIPTION
**Pull request by gallexis**

### CODE REVIEW

**Comment by tchardin** [comment]
This doesn't seem exactly right, not all CIDs will be dag-cbor, I think you might wanna check the prefix to select the right encoding no?

File: exchange/index.go [hunk 1]
```diff
@@ -546,7 +547,7 @@ func (idx *Index) CleanBlockStore(ctx context.Context) error {
 		return err
 	}
 	for k := range kc {
-		key := cid.NewCidV1(cid.DagCBOR, k.Hash())
+		key := cid.NewCidV1(cid.Raw, k.Hash())
```
Comment (tchardin) [inline exchange/index.go:550]: You shouldn't need to do this since all the keys come back with a raw codec by default

**Comment by tchardin** [comment]
@gallexis can you add little test to make sure this issue doesn't happen again? For example to detect if a future version of go-blockstore returns a different kind of CID etc. Seems like we weren't able to detect it this time

File: exchange/index_test.go [hunk 2]
```diff
@@ -658,12 +659,12 @@ func TestCleanBlockStore(t *testing.T) {
...
 
-	// generate random block1
+	// generate random block1 (codec: dagcbor)
 	blk1 := testutil.CreateRandomBlock(t, idx.Bstore())
 	require.NoError(t, idx.Bstore().Put(blk1))
 
-	// generate random block2
-	blk2 := testutil.CreateRandomBlock(t, idx.Bstore())
+	// generate random block2 (codec: raw)
+	blk2 := gstestutil.GenerateBlocksOfSize(1, 10000)[0]
```
Comment (gallexis) [inline exchange/index_test.go:667]: @tchardin GenerateBlocksOfSize() generates Raw blocks, I've tested it on the main branch and it fails but it does work on this branch

**Review by tchardin — APPROVED** [review]
LFG

**Pull request merged**